### ISSUE TITLE
Error in displaying errors?

### ISSUE THREAD

**Issue by johnmyleswhite** [JuliaLang/julia#11845]
```jl
julia> immutable Foo
       bar::NTuple{Float64, 3}
       end

julia> Foo((1., 2., 3.))
ERROR: MethodError: `convert` has no method matching convert(::Int64, ::Float64)
 in call at no fileSYSTEM: show(lasterr) caused an error
```

**Comment by yuyichao** [comment]
```julia
julia> showerror(STDERR, MethodError(convert, (3, 1.0)))
MethodError: `convert` has no method matching convert(::Int64, ::Float64)ERROR: BoundsError: attempt to access svec()
  at index [1]
 in showerror at ./replutil.jl:176
```

**Comment by yuyichao** [comment]
The printing should be fixed by #11847, which I'll probably merge after the CI's turn green.

At the same time, should we throw an error if the second parameter to `NTuple` is not a integer? Personally I get it wrong almost everytime I use `NTuple`.

**Comment by johnmyleswhite** [comment]
Thanks!
